content/blog/2017/10/vmware-disks.md

```diff
@@ -0,0 +1,391 @@
+        diskMode = '',
+        split = <unset>,
+        writeThrough = <unset>,
+        thinProvisioned = <unset>,
+        eagerlyScrub = <unset>,
+        uuid = <unset>,
+        contentId = <unset>,
+        changeId = <unset>,
+        parent = <unset>,
+        deltaDiskFormat = <unset>,
+        digestEnabled = <unset>,
+        deltaGrainSize = <unset>,
+        deltaDiskFormatVariant = <unset>,
+        sharing = <unset>,
+        keyId = <unset>
+     },
+     connectable = <unset>,
+     slotInfo = <unset>,
+     controllerKey = <unset>,
+     unitNumber = <unset>,
+     capacityInKB = 0L,
+     capacityInBytes = <unset>,
+     shares = <unset>,
+     storageIOAllocation = <unset>,
+     diskObjectId = <unset>,
+     vFlashCacheConfigInfo = <unset>,
+     iofilter = (str) [],
+     vDiskId = <unset>
+  }
+```
+
+This gives you all the available options that are needed to successfully create a new disk for a VM. Not all the information is required to be filled out, a lot of the information has the ability to be auto generated when the disk is added to a vmWare VM. Some of the options need to be whole objects as well.
+
+## Example Disk
+
+Here is an example of basic information needed to create a VMware Disk:
+
+```python
+from pyVmomi import vim
+
+disk = vim.vm.device.VirtualDisk()
+disk.backing = vim.vm.device.VirtualDisk.FlatVer2BackingInfo()
+disk.backing.diskMode = "persistent"
+disk.capacityInKB = 1024
+
+# This needs to be the current controller on the VM.
+disk.controllerKey = 1
+
+# This must be a vmWare datastore object
+disk.backing.datastore = vim.Datastore("datastore")
+
+# This number is a unique number for the disk on the current controller
+disk.unitNumber = 1
+
+output:
+  (vim.vm.device.VirtualDisk) {
+     dynamicType = <unset>,
+     dynamicProperty = (vmodl.DynamicProperty) [],
+     key = 0,
+     deviceInfo = <unset>,
+     backing = (vim.vm.device.VirtualDisk.FlatVer2BackingInfo) {
+        dynamicType = <unset>,
+        dynamicProperty = (vmodl.DynamicProperty) [],
+        fileName = '',
+        datastore = 'vim.Datastore:datastore',
+        backingObjectId = <unset>,
+        diskMode = 'persistent',
+        split = <unset>,
+        writeThrough = <unset>,
+        thinProvisioned = <unset>,
+        eagerlyScrub = <unset>,
+        uuid = <unset>,
+        contentId = <unset>,
+        changeId = <unset>,
+        parent = <unset>,
+        deltaDiskFormat = <unset>,
+        digestEnabled = <unset>,
+        deltaGrainSize = <unset>,
+        deltaDiskFormatVariant = <unset>,
+        sharing = <unset>,
+        keyId = <unset>
+     },
+     connectable = <unset>,
+     slotInfo = <unset>,
+     controllerKey = 1,
+     unitNumber = 1,
+     capacityInKB = 1024,
+     capacityInBytes = <unset>,
+     shares = <unset>,
+     storageIOAllocation = <unset>,
+     diskObjectId = <unset>,
+     vFlashCacheConfigInfo = <unset>,
+     iofilter = (str) [],
+     vDiskId = <unset>
+  }
+```
+
```

Comment:
Change to:

Notes:
 * The `disk.backing.datstore` needs to be a `pyvmomi.vim.Datastore` object for the disk to be added properly
 * The controller key can be found by querying for the current controller and getting the key ** should we show an example here** ?
 * There can be a max of 15 disks per controller.
 * The `disk.unitNumber` must in the range [0, 6] and [7, 15] (7 is reserved).
 * The `disk.unitNumber` must be unique with respect to all other disks on the controller (ex: there can only be on disk with `disk.unitNumber == 5` on a controller).